### ISSUE TITLE
ks_2samp produces only p=1 for some sample sizes

### ISSUE THREAD

**Issue by Wrzlprmft** [scipy/scipy#11507]
Consider the following minimal example:

``` Python 3
from scipy.stats import ks_2samp
import numpy as np

for _ in range(30):
    data_1 = np.random.normal(size=1000)
    data_2 = np.random.normal(size=9000)
    print( ks_2samp(data_1,data_2).pvalue )
```

This prints only `1.0`, while it should print random values uniformly distributed between 0 and 1 (because the test’s null hypothesis is true). The phenomenon is weirdly dependent on the sample size.

This issue does not seem to be in the [KS Test Master Tracker](https://github.com/scipy/scipy/issues/9797) yet. The issue appeared between 1.2.2 and 1.4.1 and came along with a strong runtime increase.

#### Scipy/Numpy/Python version information:
```
1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=5, releaselevel='final', serial=0)
```

**Comment by chrisb83** [comment]
Yes, this is a bug, see #11184. It is already fixed in master and will be part of release 1.5.0. closing this issue.

**Comment by chrisb83** [comment]
The related fix is in PR #11267
